actors/builtin/power/testing.go

```diff
@@ -0,0 +1,201 @@
+TotalQualityAdjPower abi.StoragePower
+// TotalQABytesCommitted includes claims from miners below min power threshold
+TotalQABytesCommitted abi.StoragePower
+TotalPledgeCollateral abi.TokenAmount
+
+// These fields are set once per epoch in the previous cron tick and used
+// for consistent values across a single epoch's state transition.
+ThisEpochRawBytePower     abi.StoragePower
+ThisEpochQualityAdjPower  abi.StoragePower
+ThisEpochPledgeCollateral abi.TokenAmount
+ThisEpochQAPowerSmoothed  smoothing.FilterEstimate
+
+MinerCount int64
+// Number of miners having proven the minimum consensus power.
+MinerAboveMinPowerCount int64
+
+// A queue of events to be triggered by cron, indexed by epoch.
+CronEventQueue cid.Cid // Multimap, (HAMT[ChainEpoch]AMT[CronEvent])
+
+// First epoch in which a cron task may be stored.
+// Cron will iterate every epoch between this and the current epoch inclusively to find tasks to execute.
+FirstCronEpoch abi.ChainEpoch
+
+// Claimed power for each miner.
+Claims cid.Cid // Map, HAMT[address]Claim
+
+ProofValidationBatch *cid.Cid // Multimap, (HAMT[Address]AMT[SealVerifyInfo])
+*/
+
+// Checks internal invariants of power state.
+func CheckStateInvariants(st *State, store adt.Store) (*StateSummary, *builtin.MessageAccumulator, error) {
+	acc := &builtin.MessageAccumulator{}
+
+	// basic invariants around recorded power
+	acc.Require(st.TotalRawBytePower.GreaterThanEqual(big.Zero()), "total raw power is negative %v", st.TotalRawBytePower)
+	acc.Require(st.TotalQualityAdjPower.GreaterThanEqual(big.Zero()), "total qa power is negative %v", st.TotalQualityAdjPower)
+	acc.Require(st.TotalBytesCommitted.GreaterThanEqual(big.Zero()), "total raw power committed is negative %v", st.TotalBytesCommitted)
+	acc.Require(st.TotalQABytesCommitted.GreaterThanEqual(big.Zero()), "total qa power committed is negative %v", st.TotalQABytesCommitted)
+	acc.Require(st.TotalRawBytePower.GreaterThanEqual(big.Zero()), "total raw power is negative %v", st.TotalRawBytePower)
+	acc.Require(st.TotalRawBytePower.LessThanEqual(st.TotalQualityAdjPower),
+		"total raw power %v is greater than total quality adjusted power %v", st.TotalRawBytePower, st.TotalQualityAdjPower)
+	acc.Require(st.TotalBytesCommitted.LessThanEqual(st.TotalQABytesCommitted),
+		"committed raw power %v is greater than committed quality adjusted power %v", st.TotalBytesCommitted, st.TotalQABytesCommitted)
+	acc.Require(st.TotalRawBytePower.LessThanEqual(st.TotalBytesCommitted),
+		"total raw power %v is greater than raw power committed %v", st.TotalRawBytePower, st.TotalBytesCommitted)
+	acc.Require(st.TotalQualityAdjPower.LessThanEqual(st.TotalQABytesCommitted),
+		"total qua power %v is greater than qa power committed %v", st.TotalQualityAdjPower, st.TotalQABytesCommitted)
+
+	crons, msgs, err := CheckCronInvariants(st, store)
+	acc.AddAll(msgs)
+	if err != nil {
+		return nil, acc, err
+	}
+
+	claims, msgs, err := CheckClaimInvariants(st, store)
```

Comment:
Blocking: `msgs` and `err` are both ignored here.

It looks like I've been inconsistent to date, but I think that inner methods within each actor's CheckStateInvariants should receive the accumulator as a parameter, rather than return a new one, to avoid this easy-to-make mistake.